lib/src/utils/system.dart

```diff
@@ -14,12 +14,11 @@ class System {
   static const instance = System._();
 
   /// Returns the system total RAM in megabytes.
-  Future<int?> getTotalRam() async {
+  Future<int> getTotalRam() async {
     try {
-      return await _channel.invokeMethod<int>('getTotalRam');
-    } on PlatformException catch (e) {
-      debugPrint('Failed to get total RAM: ${e.message}');
```

Comment:
Why this change? Now if the RAM cannot be retrieved it fails silently.

Reply:
i should have changed it back its just an artifact from testing i was doing.
basically method channels dont work on the background isolate and it was causing the `appInitializationProvider` to fail 
i changed it back and added a new check for the error that occurs when you call a method channel on the background isolate